src/Symfony/Component/HttpKernel/Event/ControllerArgumentsEvent.php

```diff
@@ -94,10 +94,16 @@ public function getNamedArguments(): array
     }
 
     /**
-     * @return array<class-string, list<object>>
+     * @template T of class-string|null
+     *
+     * @param T $className
+     *
+     * @return array<class-string, list<object>>|list<object>
+     *
+     * @psalm-return (T is null ? array<class-string, list<object>> : list<object>)
```

Comment:
Instead of a conditional return type, I'm wondering whether a separate method would be better.

Reply:
We already have similar APIs and they work great IMHO.